File: ssm-chaincode/ssm-chaincode-client/src/test/java/ssm/chaincode/client/SsmClientOptionalTest.java
package ssm.chaincode.client;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import ssm.chaincode.dsl.SsmAgentBase;
import ssm.chaincode.dsl.SsmBase;
import ssm.chaincode.dsl.SsmSessionStateBase;

public class SsmClientOptionalTest {

    private static SsmClient client;

    @BeforeAll
    public static void init() throws Exception {
        client = SsmClientTestBuilder.build();
    }

    @Test
    public void getAdminUser() throws Exception {
        CompletableFuture<Optional<SsmAgentBase>> agentRet = client.getAdmin(UUID.randomUUID().toString());
        Optional<SsmAgentBase> agentFormClient = agentRet.get();
        assertThat(agentFormClient).isEmpty();
    }

    @Test
    public void getAgentUser2() throws Exception {
        CompletableFuture<Optional<SsmAgentBase>> agentRet = client.getAgent(UUID.randomUUID().toString());
        Optional<SsmAgentBase> agentFormClient = agentRet.get();
        assertThat(agentFormClient).isEmpty();
    }

    @Test
    public void getSsm() throws Exception {
        CompletableFuture<Optional<SsmBase>> ssmReq = client.getSsm(UUID.randomUUID().toString());
        Optional<SsmBase> ssm = ssmReq.get();
        assertThat(ssm).isEmpty();
    }

    @Test
    public void getSession() throws Exception {
        CompletableFuture<Optional<SsmSessionStateBase>> ses = client.getSession(UUID.randomUUID().toString());
        Optional<SsmSessionStateBase> sesReq = ses.get();
        assertThat(sesReq).isEmpty();
    }

}

File: ssm-chaincode/ssm-chaincode-client/src/test/kotlin/ssm/chaincode/client/SsmClientOptionalTest.kt
package ssm.chaincode.client

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import ssm.chaincode.client.SsmClient
import java.util.*

class SsmClientOptionalTest {
    @get:Throws(Exception::class)
    @get:Test
    val adminUser: Unit
        get() {
            val agentRet = client!!.getAdmin(UUID.randomUUID().toString())
            val agentFormClient = agentRet.get()
            Assertions.assertThat(agentFormClient).isNotNull
        }

    @get:Throws(Exception::class)
    @get:Test
    val agentUser2: Unit
        get() {
            val agentRet = client!!.getAgent(UUID.randomUUID().toString())
            val agentFormClient = agentRet.get()
            Assertions.assertThat(agentFormClient).isNotNull
        }

    @get:Throws(Exception::class)
    @get:Test
    val ssm: Unit
        get() {
            val ssmReq = client!!.getSsm(UUID.randomUUID().toString())
            val ssm = ssmReq.get()
            Assertions.assertThat(ssm).isNotNull
        }

    @get:Throws(Exception::class)
    @get:Test
    val session: Unit
        get() {
            val ses = client!!.getSession(UUID.randomUUID().toString())
            val sesReq = ses.get()
            Assertions.assertThat(sesReq).isNotNull
        }

    companion object {
        private var client: SsmClient? = null
        @BeforeAll
        @Throws(Exception::class)
        fun init() {
            client = SsmClientTestBuilder.build()
        }
    }
}